crates/datasources/src/bson/mod.rs

```diff
@@ -45,13 +45,16 @@ use datafusion::arrow::datatypes::{
     TimestampNanosecondType,
 };
 use datafusion::arrow::error::ArrowError;
+use datafusion::arrow::record_batch::RecordBatch;
 
 pub struct BsonBatchConverter {
     batch: StructArray,
     schema: Vec<String>,
     row: usize,
     started: bool,
     columns: Vec<Vec<bson::Bson>>,
+    src_err: Option<ArrowError>,
+    override_id: bool,
```

Comment:
i think adding a small comment as to what this does will be helpful for the future.

At first glance it's not really self explanatory that this flag will override the mongo ids if set, or when it should even be set.

Reply:
removed